microsoft-edge/progressive-web-apps-chromium/how-to/widgets.md

```diff
@@ -0,0 +1,531 @@
+
+For the time being, only custom Adaptive Cards templates are supported. Adaptive Cards are an open card exchange format that can be used to exchange UI content in a common and consistent way. See [Adaptive Cards Overview](/adaptive-cards/).
+
+To define a custom Adaptive Cards template, use the `ms_ac_template` field:
+
+```json
+{
+  "name": "Agenda",
+  "description": "Your day, at a glance",
+  "tag": "agenda",
+  "data": "/widgets/data/agenda",
+  "type": "application/json",
+  "template": "calendar-agenda",
+  "ms_ac_template": "/widgets/template/agenda_template.ac.json"
+}
+```
+
+The `ms_ac_template` field value should be a valid URL to a template file.
+
+Here is an example of an Adaptive Cards template:
+
+```json
+{
+  "type": "AdaptiveCard",
+  "body": [
+    {
+      "type": "TextBlock",
+      "size": "Medium",
+      "text": "Hello ${name}"
+    },
+    {
+      "type": "TextBlock",
+      "text": "You have ${meetingNb} meetings today"
+    }
+  ],
+  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+  "version": "1.5"
+}
+```
+
+To learn more, see [Adaptive Cards Templating](/adaptive-cards/templating/).
+
+
+<!-- ====================================================================== -->
+## Bind data to your template
+
+The template declares the user interface of a Widget.  Data then populates this user interface.
+
+To bind data to your template, use the `data` field in your Widget definition. This field should be set to a URL that returns valid JSON data.
+
+The template defined in [the previous section](#define-a-widget-template) contains two variables: `name` and `meetingNb`, which are enclosed in the binding expression syntax: `${}`. The data that's returned by the `data` URL in your Widget definition should contain values for these variables.
+
+Here is an example of what the `data` URL might return: 
+
+```json
+{
+  "name": "Amelie Garner",
+  "meetingNb": 3
+}
+```
+
+
+<!-- ====================================================================== -->
+## Define Widget actions
+
+If you want your Widget to let users perform tasks, use the `actions` field in your Widget definition, and define a template that supports actions.
+
+The `actions` field is an array of action definitions. Here is an example:
+
+```json
+{
+  "name": "Agenda",
+  "description": "Your day, at a glance",
+  "tag": "agenda",
+  "data": "/widgets/data/agenda",
+  "type": "application/json",
+  "ms_ac_template": "/widgets/template/agenda_template.ac.json",
+  "actions": [
+    {
+      "action": "create-meeting",
+      "title": "Create a new meeting"
+    }
+  ]
+}
+```
+
+Use the `action` field to give the action a name. This name will be used to handle the action in your service worker. The `title` field is presented to the user.
+
+To actually trigger the action from the Widget, your template needs to support it. Here is an example of an action defined in a custom Adaptive Cards template:
```

Comment:
All the widget action content above this line is no longer needed, but this line and everything below are still valid